brave/brave-ios
# Fix #7723: Selected Account Refactor & Bump BraveCore to v1.58.26

## Summary of Changes
- Address changes for the Selected Account Refactor in BraveCore.
- `GetSelectedCoin()` was removed. In most places we were fetching the selected coin so we knew which account or network to fetch. 
    - We now use `GetAllAccounts()` on KeyringService which returns a `selectedAccount` which we can get it's `CoinType` from. In the cases where we were previously fetching coin to get the account, this works well. In a couple spots we now call `GetAllAccounts()` to get the `selectedAccount` coin type so we can fetch the currently selected network (ex. `BuyTokenStore` now hold a reference to KeyringService for this purpose).
    - This method will also return the `ethDappSelectedAccount`/`solDappSelectedAccount` account for use with DApps.

This pull request fixes #7723

## Submitter Checklist:


## Test Plan:
The underlying core change refactored how selected account's work. This touched every view that uses a selected account, so requires a bit of regression testing.
- Wallet Panel
    - Verify switching selected account via the panel is still working as expected. If changing between Ethereum / Solana account, the network must also update.
    - Verify switching network via the panel is still working as expected. If changing between Ethereum / Solana networks, the account must also update
- DApp connections
    - Verify the selected account is selected by default when connecting to an Etheruem DApp for the first time. ex. have Ethereum Account 2 selected in the wallet panel, then on the Ethereum DApp tap 'connect wallet'. Ethereum Account 2 should be selected by default.
- Buy Token View
    - Verify switching selected account, verify switching selected network
- Send Token View
    - Verify selecting a token switches to the correct network and account. Test switching from EthAccount1 -> SolAccount1 -> EthAccount2 for example.
- Swap Token View
    - Verify switching selected account, verify switching selected network
- Asset Detail View
    - Verify 'send' button on asset detail screens will open with the expected token selected by default.
    - Verify 'swap' button on asset detail screens will open with the expected token selected by default.
- Add Account
    - Verify you can add an account for both Ethereum & Solana.


## Screenshots:



## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
